cli/azd/cmd/cmd_help.go

```diff
@@ -0,0 +1,339 @@
+	cmdGroupManage commandGroupAnnotationValue = commandGroupAnnotationValue(i18nCmdGroupTitleManage)
+	// cmdGroupConfig defines a group of commands for Monitoring.
+	cmdGroupMonitor commandGroupAnnotationValue = commandGroupAnnotationValue(i18nCmdGroupTitleMonitor)
+	// cmdGroupConfig defines a group of commands for Help and About.
+	cmdGroupAbout commandGroupAnnotationValue = commandGroupAnnotationValue(i18nCmdGroupTitleAbout)
+)
+
+// getGroupCommandAnnotation check if there is a grouping annotation for the command. Returns the annotation value as an
+// i18nTextId (so it can be used directly to resolve a string) if the annotation is found. Otherwise, returns `"", false` to
+// indicate the command has no grouping annotation.
+func getGroupCommandAnnotation(cmd *cobra.Command) (i18nTextId, bool) {
+	annotationValue, found := cmd.Annotations[string(cmdGrouperKey)]
+	return i18nTextId(annotationValue), found
+}
+
+// cmdHelpGenerator defines the required signature to implement and produce help description for commands.
+type cmdHelpGenerator func(cmd *cobra.Command) string
+
+// generateCmdHelpOptions defines settings to control the text description for displaying the commands' help.
+type generateCmdHelpOptions struct {
+	Description cmdHelpGenerator
+	Usage       cmdHelpGenerator
+	Commands    cmdHelpGenerator
+	Flags       cmdHelpGenerator
+	Footer      cmdHelpGenerator
+}
+
+/*
+generateCmdHelp sets the base structure for displaying help documentation for a command in the console.
+The base structure is on the form of:
+
+**********************
+
+<description>
+
+<usage>
+<commands>
+<flags>
+<footer>
+
+**********************
+
+Where:
+  - description: Main information for the command. Default to cobra's `Short` field.
+  - usage: Demonstrate how to call the command. Default to cobra's `Use` filed.
+  - commands: The list of sub-commands supported. Default to list cobra's sub-commands in the form of `cmd : short-notes`.
+  - flags: List of supported flags. Default to `Flags + Global flags` and each flag as `-F, --flag [type] : description`.
+  - footer: The last section is where commands can define quick-start, examples or extra notes. Default to display notes
+    about how to report bugs or comments.
+*/
+func generateCmdHelp(
+	cmd *cobra.Command,
+	options generateCmdHelpOptions) string {
+
+	getGeneratorOrDefault := func(option, defaultOption cmdHelpGenerator) cmdHelpGenerator {
+		if option != nil {
+			return option
+		}
+		return defaultOption
+	}
+
+	return fmt.Sprintf("\n%s%s%s%s%s%s\n",
+		getGeneratorOrDefault(options.Description, getCmdHelpDefaultDescription)(cmd),
+		getGeneratorOrDefault(options.Usage, getCmdHelpDefaultUsage)(cmd),
+		getGeneratorOrDefault(options.Commands, getCmdHelpDefaultCommands)(cmd),
+		getGeneratorOrDefault(options.Flags, getCmdHelpDefaultFlags)(cmd),
+		getPreFooter(cmd),
+		getGeneratorOrDefault(options.Footer, getCmdHelpDefaultFooter)(cmd),
+	)
```

Comment:
Have you considered leveraging go templates more in this feature? We could introduce a go schema for the structure of a `help doc` and use our own HTML style markup for various different formats that then gets replaced at runtime with the correct CLI formatting.

This would reduce the special case styling we need to use for the underlines, colors and any other formatting.

Reply:
yes, that would require implementing a `string resolver` which knows how to handle the format we ask in the templates. For example ` {{ .value | withBold | withUnderline }}`  , then, the `string resolver` needs a reference about what `withBold` and `withUnderline` means, which is basica a function call to the logic we currently have.
So, at the end, we would be using the current functions we have for formatting, but trough a `string resolver`

Reply:
I thought we were going to support go templates?  It would allow for more direct verbose support for anyone going in here to manage the docs.

Reply:
I don't think is worth it to create a new template resolver just for this. But we can add it to the list of things to review and re-iterate in the next sprints